recipes/calceph/all/conanfile.py

```diff
@@ -42,27 +44,38 @@ def configure(self):
         del self.settings.compiler.libcxx
         if self.settings.compiler == "Visual Studio":
             del self.options.threadsafe
-            if self.options.shared:
-                raise ConanInvalidConfiguration("calceph doesn't support shared builds with Visual Studio yet")
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            raise ConanInvalidConfiguration("calceph doesn't support shared builds with Visual Studio yet")
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        if tools.os_info.is_windows and self.settings.compiler != "Visual Studio" and \
-           "CONAN_BASH_PATH" not in os.environ:
-            self.build_requires("msys2/20200517")
+        if self._settings_build.os == "Windows" and self.settings.compiler != "Visual Studio" and \
+           not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename(self.name + "-" + self.version, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    @contextmanager
+    def _msvc_build_environment(self):
+        with tools.chdir(self._source_subfolder):
```

Comment:
Doesn't this belong in `build` and `package`?
Changing the working directory in a contextmanager is obfuscating the working directory.

Reply:
ok, fixed